app/models/casebook/concerns/contents.rb

```diff
@@ -0,0 +1,53 @@
+module Casebook::Concerns::Contents
+  extend ActiveSupport::Concern
+
+  def children
+    contents.where(['array_length(casebooks.ordinals, 1) = ?', ordinals.length + 1])
```

Comment:
Yeah this is confusing! Sorry I didn't write anything up about it, It's one of the things I was meaning to do when I got sick D:

"Ordinal" is just a fancy word for ordering numbers like "first", "second", "third", etc. Another name for this column could be "path"; it just an array of integers which stores the full position of this section in the main table of contents. 

So the third resource in the second part of the fourth chapter has a path of [4, 2, 3].

This column is indexed, so all sections of a casebook can be looked up by the combination of their Book's (root) ID and their ordinals. The upshot of this is you can easily locate a section from a descriptive URL like `/casebooks/100-third-restatement-torts/sections/4.3.2-negligence`

It does add some weight to tasks like reordering of sections, so it's something I'd love to get your intuition on, but I feel like it's a simpler model overall, since each piece of content in a casebook can be looked at in isolation.